pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -738,63 +740,43 @@ func (volTopology *nodeVolumeTopology) GetNodeTopologyLabels(ctx context.Context
 	log := logger.GetLogger(ctx)
 
 	var err error
-	if volTopology.isCSINodeIdFeatureEnabled && volTopology.clusterFlavor == cnstypes.CnsClusterFlavorVanilla {
-		csiNodeTopology := &csinodetopologyv1alpha1.CSINodeTopology{}
-		csiNodeTopologyKey := types.NamespacedName{
-			Name: nodeInfo.NodeName,
-		}
-
-		// Get CsiNodeTopology instance
-		err = volTopology.csiNodeTopologyK8sClient.Get(ctx, csiNodeTopologyKey, csiNodeTopology)
-		if err != nil {
-			if apierrors.IsNotFound(err) {
-				err = createCSINodeTopologyInstance(ctx, volTopology, nodeInfo)
-				if err != nil {
-					return nil, logger.LogNewErrorCodef(log, codes.Internal, err.Error())
-				}
-			} else {
-				msg := fmt.Sprintf("failed to get CsiNodeTopology for the node: %q. Error: %+v", nodeInfo.NodeName, err)
-				return nil, logger.LogNewErrorCodef(log, codes.Internal, msg)
-			}
-		} else {
-			// If CSINodeTopology instance already exists, check if the NodeUUID
-			// parameter in Spec is populated. If not, patch the instance.
-			if csiNodeTopology.Spec.NodeUUID == "" ||
-				csiNodeTopology.Spec.NodeUUID != nodeInfo.NodeID {
-				if csiNodeTopology.Spec.NodeUUID == "" {
-					log.Infof("CSINodeTopology instance: %q with empty nodeUUID found. "+
-						"Patching the instance with nodeUUID", nodeInfo.NodeName)
-				} else {
-					log.Infof("CSINodeTopology instance: %q with different "+
-						"nodeUUID: %s found. Patching the instance with nodeUUID: %s",
-						nodeInfo.NodeName, csiNodeTopology.Spec.NodeUUID, nodeInfo.NodeID)
-				}
-				patch := []byte(fmt.Sprintf(`{"spec":{"nodeID":"%s","nodeuuid":"%s"}}`, nodeInfo.NodeName, nodeInfo.NodeID))
-				// Patch the CSINodeTopology instance with nodeUUID
-				err = volTopology.csiNodeTopologyK8sClient.Patch(ctx,
-					&csinodetopologyv1alpha1.CSINodeTopology{
-						ObjectMeta: metav1.ObjectMeta{
-							Name: nodeInfo.NodeName,
-						},
-					},
-					client.RawPatch(types.MergePatchType, patch))
-				if err != nil {
-					msg := fmt.Sprintf("Fail to patch CsiNodeTopology for the node: %q "+
-						"with nodeUUID: %s. Error: %+v",
-						nodeInfo.NodeName, nodeInfo.NodeID, err)
-					return nil, logger.LogNewErrorCodef(log, codes.Internal, msg)
-				}
-				log.Infof("Successfully patched CSINodeTopology instance: %q with Uuid: %q",
-					nodeInfo.NodeName, nodeInfo.NodeID)
-			}
+	csiNodeTopology := &csinodetopologyv1alpha1.CSINodeTopology{}
```

Comment:
Can we move `volTopology.clusterFlavor == cnstypes.CnsClusterFlavorVanilla` check before line 743 as we did not run this code for Guest cluster before.

Reply:
That should not matter IMO. We are still not updating or doing anything on `Guest` clusters, other than creating the `CSINodeTopology` if it does not exist(which we did before as well).  The logic is still the same.

What I have tried to do here is - to remove lots of deep nested if/elses by using early returns in case of errors and other cases. So logic now goes:

1. Can we find `CSINodeTopology`, if no go to step 2.
2. Create a new `CSINodeTopology` and wait for it to have `success`  or `error` state.
3. If we can find `CSINodeTopology` then 
4. On vanilla clusters
5. If use `CSINodeID` feature, update the `nodeid` for topology
6. Make sure `status` is reset back to empty string. patch the object and wait for update.
7.

Logic before is:
1. Are we on vanilla clusters and using `CSINOdeID` feature?
2. If no - create `CSINodeTopology` and wait for it to have `success` or `error` state.
3. If yes - can we find existing `CSINodeTopology` object?
4. If `CSINodeTopology` already exists then update `nodeID` if required. Patch the object and wait for it.
5. If `CSINodeTopology` does not exist then create new `CSINodeTopology` and wait for it to have `success` or `error` state.

Correct me if I am wrong, but end result is same as before.  Code is somewhat cleaner and I can use a common patch to update `nodeID` and reset `status` field in same patch request.